lib/adequate_crypto_address/cryptos/ada.rb

```diff
@@ -0,0 +1,32 @@
+module AdequateCryptoAddress
+  class Ada
+    attr_reader :address, :type
+
+    def initialize(address_string)
+      @address = address_string
+      @type = address_type
+    end
+
+    def valid?(address)
+      if !valid_prefix?
+        false
+      else
+        raise "Invalid address: #{address}" unless valid_prefix?
+      end
+    end
```

Comment:
Both of these conditions are the same and never returning `true`.

I don't think we should be raising an error on a predicate method. It should just return `true`/`false`. Since `valid_prefix?` is doing the validation, you can do: 

`alias valid? valid_prefix?`